File: todo-backend/src/main/java/com/example/todo_backend/services/CardService.java
package com.example.todo_backend.services;

import java.util.List;

import com.example.todo_backend.dtos.CardDTO;

public interface CardService {
    CardDTO createCard(CardDTO dto, Long userId);
    CardDTO updateCard(Long cardId, CardDTO cardDTO);

    List<CardDTO> getCardsByListId(Long listId);
    void deleteCard(Long id);
}

File: todo-frontend/src/app/core/services/card.service.ts
import { Injectable, inject } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { environment } from '../../../environments/environment';
import { CardDTO } from '../models/card.model';

@Injectable({ providedIn: 'root' })
export class CardService {
  private http = inject(HttpClient);

  getCards(listId: number) {
    return this.http.get<CardDTO[]>(`${environment.apiUrl}/cards/getCardsByList/${listId}`);
  }

  createCard(payload: Partial<CardDTO>) {
    return this.http.post<CardDTO>(`${environment.apiUrl}/cards/createCard`, payload);
  }
  updateCard(cardId: number,payload: Partial<CardDTO>) {
    return this.http.put<CardDTO>(`${environment.apiUrl}/cards/updateCard/${cardId}`, payload);
  }

  deleteCard(cardId: number) {
    return this.http.delete(`${environment.apiUrl}/cards/deleteCard/${cardId}`);
  }
}
